src/utils/draftbots.js

```diff
@@ -104,29 +104,47 @@ const scaleSimilarity = (value) =>
 const SYNERGY_SCALE = 0.2;
 const MAX_SYNERGY = 10;
 
-let synergyMatrix;
+const xorStrings = (a, b) => {
+  let s = '';
 
-export const setSynergyMatrix = (size) => {
-  synergyMatrix = [];
-  for (let i = 0; i <= size; i++) {
-    synergyMatrix.push(new Array(size + 1).fill(null));
+  // use the longer of the two words to calculate the length of the result
+  for (let i = 0; i < Math.max(a.length, b.length); i++) {
+    // append the result of the char from the code-point that results from
+    // XORing the char codes (or 0 if one string is too short)
+
+    // eslint-disable-next-line no-bitwise
+    s += String.fromCharCode((a.charCodeAt(i) || 0) ^ (b.charCodeAt(i) || 0));
```

Comment:
This should use a better hash, this is relatively likely to collide. Honestly I'd prefer to just use the names as keys directly after some minor sanitizing such as replacing illegal characters with something like `\u<CODEPOINT>` and adding a `__`(double underscore) between the two names. Could also just go back to using indices as well which is probably the fastest way to do it, if you don't want to store the whole matrix then use `<index1>_<index2>` as your keys, but this matrix isn't that big (typical draft has at most 800 cards which is only 5mb when stored as a complete matrix of doubles).